### PR TITLE
[formrecognizer] reduce time for recorded tests runs

### LINKED ISSUE #11932: [formrecognizer] look into speeding up recorded runs
recorded test runs can take up to 30 mins. See if we can play around with the polling interval to shorten run times.

Also refactor how we're doing the delay between account creation and training in order to skip the sleep for recorded runs.

### PR DESCRIPTION
**Pull request by kristapratico**
Resolves #11932 

Reduce the time our recorded tests runs take by setting `polling_interval=0` if not running live.
Pipeline run times decrease from about 35 mins to 8 mins with this PR.

This PR ended up with a few things:
* Refactor testcase.py and add `GlobalClientPreparer` - this is so we can create the client for each test in the preparer and set the polling_interval accordingly here instead of passing that keyword into each method in each test
* Sets the default polling interval (5s) at the client-level now, and allows user override at the client or operation-level
* Adds tests for polling interval for FormRecognizerClient and FormTrainingClient
* Adds a transport wrapper for the `get_client` method b/c we weren't sharing the pipeline between the clients previously. @rakshith91 - can you look at this part? I remember you doing this work for storage as well.

Live test pass: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=422700&view=results

### CODE REVIEW

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py [hunk 1]
```diff
@@ -66,11 +66,13 @@ def __init__(self, endpoint, credential, **kwargs):
         # type: (str, Union[AzureKeyCredential, TokenCredential], Any) -> None
 
         authentication_policy = get_authentication_policy(credential)
+        polling_interval = kwargs.pop("polling_interval", POLLING_INTERVAL)
```
Comment (iscai-msft) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py:69]: Are we documenting this `polling_interval` kwarg to the client for users? I think a good idea could be to call it `_polling_interval` if we're the only people expecting the use it, but I don't feel too strongly about this

Comment (kristapratico) [reply sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py:69]: This is an azure-core setting, so hopefully once Xiang documents all the core keywords we'll be able to link to that :)

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_training_client.py [hunk 2]
```diff
@@ -371,11 +374,20 @@ def get_form_recognizer_client(self, **kwargs):
...
-            endpoint=self._endpoint,
-            credential=self._credential,
+
+        _pipeline = Pipeline(
+            transport=TransportWrapper(self._client._client._pipeline._transport),
+            policies=self._client._client._pipeline._impl_policies
+        )  # type: Pipeline
+        client = FormRecognizerClient(
+            self._endpoint,
+            self._credential,
```
Comment (rakshith91) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_training_client.py:384]:
```suggestion
            endpoint=self._endpoint,
            crendential=self._credential,
```

**Comment by kristapratico** [comment]
> Looks good - but please add tests for the transport wrapper.
> 
> You can refer these:
> 
> https://github.com/Azure/azure-sdk-for-python/blob/379ae6e21a6abb3f8a134bf610c77e5f9e697786/sdk/storage/azure-storage-blob/tests/test_common_blob.py#L2005

we do have tests, discussed offline

**Pull request merged**